packages/api/src/services/github/api.ts

```
@@ -0,0 +1,52 @@
+import axios from "axios";
+
+const GITHUB_API_DOMAIN = "https://api.github.com";
+
+interface OrganizationsDTO {
+  login: string;
+  description: string;
+  id: string;
+  avatar_url: string;
+  repos_url: string;
+}
+
+export interface OrganizationData {
+  name: string;
+  id: string;
+  description: string;
+  avatar: string;
+  repos: string;
+}
+
+const sanitizeOrganizationData = (
+  orgs: OrganizationsDTO[]
+): OrganizationData[] => {
+  const sanitizedOrganizations: OrganizationData[] = orgs.map((org) => ({
+    name: org.login,
+    description: org.description,
+    id: org.id,
+    avatar: org.avatar_url,
+    repos: org.repos_url,
+  }));
+  return sanitizedOrganizations;
+};
+
+export const fetchOrganizations = async (token: string) => {
+  try {
+    const options = {
+      headers: {
+        Authorization: `token ${token}`,
+      },
+    };
+    const response = await axios.get(GITHUB_API_DOMAIN + "/user/orgs", options);
+
+    if ("error" in response.data) {
+      console.error(response.data.error);
+      return null;
```

why not surface the error to the client?

---

At first I thought that, if there's an error from github, the client shouldn't be aware of it... But now you are making me think 🤔  Maybe we should send the error to the client indeed

---

I'm a proponent of being very transparent with errors since it makes debugging easier for us and the client

---

Related to: https://github.com/Web3-API/web3hub/issues/13